android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java

```diff
@@ -1176,6 +1186,59 @@ public void onBackPressed() {
         super.onBackPressed();
     }
 
+    private void onResolveWalletAddressDone(String domain, String result) {
+        if (!domain.equals(mSendToAddrText.getText().toString())) {
+            return;
+        }
+
+        if (result == null || result.isEmpty()) {
+            mResolvedAddrText.setVisibility(View.GONE);
+            mResolvedAddrText.setText("");
+            String notRegsteredErrorText = String.format(
+                    getString(R.string.wallet_domain_not_registered_error_text), domain);
+
+            setSendToFromValueValidationResult(notRegsteredErrorText, true, true);
+        } else {
+            mResolvedAddrText.setVisibility(View.VISIBLE);
+            mResolvedAddrText.setText(result);
+            setSendToFromValueValidationResult("", false, true);
+        }
+    }
+
+    private boolean maybeResolveWalletAddress() {
+        String domain = mSendToAddrText.getText().toString();
+
+        if (WalletNativeUtils.isUnstoppableDomainsTld(domain)) {
+            mJsonRpcService.unstoppableDomainsGetWalletAddr(
+                    domain, mCurrentBlockchainToken, (response, errorResponse, errorString) -> {
+                        onResolveWalletAddressDone(domain, response);
+                    });
+            return true;
+        }
+
+        if (mCurrentBlockchainToken.coin == CoinType.ETH && WalletNativeUtils.isEnsTld(domain)) {
+            mJsonRpcService.ensGetEthAddr(domain, null,
+                    (response, requireOffchainConsent, errorResponse, errorString) -> {
+                        onResolveWalletAddressDone(domain, response);
+                    });
+            return true;
+        }
+
+        if (ChromeFeatureList.isEnabled(BraveFeatureList.BRAVE_WALLET_SNS)) {
+            if (mCurrentBlockchainToken.coin == CoinType.SOL
+                    && WalletNativeUtils.isSnsTld(domain)) {
+                mJsonRpcService.snsGetSolAddr(domain, (response, errorResponse, errorString) -> {
+                    onResolveWalletAddressDone(domain, response);
+                });
+                return true;
+            }
+        }
+
+        mResolvedAddrText.setVisibility(View.GONE);
+        mResolvedAddrText.setText("");
+        return false;
+    }
```

Comment:
Eventually will be refactored to a single mojo call https://github.com/brave/brave-browser/issues/26655